partcad/src/partcad/sentry.py

```
@@ -0,0 +1,135 @@
+        yield span
+
+
+def start_span_as(name, category: str = ""):
+    def decorator(func):
+        def wrapper(*args, **kwargs):
+            parent = trace.get_current_span()
+            tag = name if not category else f"{category}.{name}"
+            if getattr(parent, "tag", "") == tag:
+                return func(*args, **kwargs)
+            with tracer.start_as_current_span(tag) as span:
+                setattr(span, "tag", tag)
```

Ruff (0.8.2)

Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)